### PR TITLE
libraries: closed date API

### PR DESCRIPTION
**Pull request by zannkukai**
Creates an simple API to know all closed dates between tow dates for a
specific libraries.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## How to test?

- `api/library/1/closed_dates`

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/libraries/api_views.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
```
Comment (lauren-d) [inline rero_ils/modules/libraries/api_views.py:4]: 2021 RERO
2021 UCLouvain?

File: rero_ils/modules/libraries/api_views.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+@api_blueprint.route('/<library_pid>/closed_dates', methods=['GET'])
+@check_logged_as_librarian
+def list_closed_dates(library_pid):
+    """HTTP GET request to get the closed date about a library."""
+    library = Library.get_record_by_pid(library_pid)
+    if not library:
+        abort(404)
+
+    # get start date from 'from' parameter from query string request
+    sdate = request.args.get('from', datetime.now() - timedelta(days=31))
```
Comment (lauren-d) [inline rero_ils/modules/libraries/api_views.py:46]: COSMETIC: It's more readable to use start_date instead sdate (same for edate and tdate).

**Review by benerken — APPROVED** [review]

File: rero_ils/modules/libraries/api_views.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    'api_library',
+    __name__,
+    url_prefix='/library'
+)
+
+
+@api_blueprint.route('/<library_pid>/closed_dates', methods=['GET'])
+@check_logged_as_librarian
+def list_closed_dates(library_pid):
+    """HTTP GET request to get the closed date about a library."""
```
Comment (BadrAly) [inline rero_ils/modules/libraries/api_views.py:41]: `..dates for a given library_pid`

Comment (BadrAly) [inline rero_ils/modules/libraries/api_views.py:41]: Can you add some description in the method about the `from` and `until` arguments?

File: rero_ils/modules/libraries/api_views.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    if not library:
+        abort(404)
+
+    # get start date from 'from' parameter from query string request
+    start_date = request.args.get('from', datetime.now() - timedelta(days=31))
+    if isinstance(start_date, str):
+        start_date = date_string_to_utc(start_date)
+    start_date = start_date.replace(tzinfo=library.get_timezone())
+    # get end date from 'until' parameter from query string request
+    end_date = request.args.get('until', add_years(datetime.now(), 1))
```
Comment (BadrAly) [inline rero_ils/modules/libraries/api_views.py:52]: does this mean you return the closed dates of one complete year if the `until` is not given?

**Review by BadrAly — CHANGES_REQUESTED** [review]

File: rero_ils/modules/libraries/api_views.py (hunk 4 above)
Comment (zannkukai) [reply rero_ils/modules/libraries/api_views.py:52]: exactly : `from` default value are now()-1month, `until` default value are now()+1year. It seems suffisent in most of the case to disable off-day in the datepicker UI component.

**Review by BadrAly — APPROVED** [review]

**Pull request merged**